src/Ledger/Ratify.lagda

```diff
@@ -263,39 +266,83 @@ actualVotes Γ pparams cc ga votes
   roleVotes r = mapKeys (uncurry credVoter) (filter (λ (x , _) → r ≡ proj₁ x) votes)
 
   activeDReps = dom (filter (λ (_ , e) → currentEpoch ≤ e) dreps)
-  spos = filterˢ IsSPO (dom (stakeDistr stakeDistrs))
+  spos  = filterˢ IsSPO (dom (stakeDistr stakeDistrs))
+\end{code}
+\begin{code}
 
   getCCHotCred : Credential × Epoch → Maybe Credential
   getCCHotCred (c , e) = case ¿ currentEpoch ≤ e ¿ᵇ , lookupᵐ? ccHotKeys c of
 \end{code}
 \begin{code}[hide]
-    λ where
+      λ where
+\end{code}
+\begin{code}
+        (true , just (just c'))  → just c'
+        _                        → nothing -- expired, no hot key or resigned
+\end{code}
+\begin{code}[hide]
+  getPoolParams : Credential → Maybe PoolParams
+  getPoolParams (KeyHashObj kh) = lookupᵐ? pools kh
+  getPoolParams _ = nothing
+\end{code}
+\begin{code}
+
+  SPODefaultVote : GovAction → VDeleg → Vote
+  SPODefaultVote (TriggerHF _) _ = Vote.no
+  SPODefaultVote NoConfidence (credVoter SPO c) = case getPoolParams c of
+\end{code}
+\begin{code}[hide]
+      λ where
+\end{code}
+\begin{code}
+        nothing → Vote.no
+        (just  p) → case lookupᵐ? delegatees (PoolParams.rewardAddr p) of
+\end{code}
+\begin{code}[hide]
+               λ where
+\end{code}
+\begin{code}
+               (just noConfidenceRep)  → Vote.yes
+               (just abstainRep)       → Vote.abstain
+               _                       → Vote.no
+  SPODefaultVote _ (credVoter SPO c) = case getPoolParams c of
+\end{code}
+\begin{code}[hide]
+      λ where
+\end{code}
+\begin{code}
+        nothing → Vote.no
```

Comment:
Same as above

Reply:
Still working to get, simultaneously, decent indentation, hidden `λ where`s, and no TeX compilation error (`! Package array Error: Empty preamble: l' used.`).

Reply:
Apparently TeX is extremely sensitive to the indentation used after hidden `λ where`s and especially nested hidden `λ where`s, but I managed to find an indentation that compiles and looks reasonable.

Reply:
Yeah, this is all held together with duct tape